frontend/app/src/components/wallet-bridge/Pairing.vue

```diff
@@ -0,0 +1,345 @@
+<script setup lang="ts">
+import type { SessionTypes } from '@walletconnect/types';
+import type { TransactionRequest, TransactionResponse } from 'ethers';
+import { useSupportedChains } from '@/composables/info/chains';
+import { useWalletHelper } from '@/composables/trade/wallet-helper';
+import { EIP155, ROTKI_DAPP_METADATA, useWalletStore } from '@/store/trade/wallet';
+import { type IWalletKit, WalletKit, type WalletKitTypes } from '@reown/walletkit';
+import { assert } from '@rotki/common';
+import { get, set } from '@vueuse/core';
+import { Core } from '@walletconnect/core';
+import { formatJsonRpcError } from '@walletconnect/jsonrpc-utils';
+import { buildApprovedNamespaces, getSdkError } from '@walletconnect/utils';
+import { onMounted, ref } from 'vue';
+
+interface LogEntry {
+  message: string;
+  timestamp: string;
+  type: 'info' | 'success' | 'error';
+}
+
+const props = defineProps<{
+  address: string;
+  connectedChainId: number;
+}>();
+
+const { address, connectedChainId } = toRefs(props);
+
+const { t } = useI18n();
+const projectId = import.meta.env.VITE_WALLET_CONNECT_PROJECT_ID as string;
+
+const core = new Core({
+  projectId,
+});
+
+const pairUri = ref('');
+const logs = ref<LogEntry[]>([]);
+const isConnecting = ref(false);
+const walletKit = ref<IWalletKit | null>(null);
+const activeSessions = ref<SessionTypes.Struct[]>([]);
+
+const { getBrowserProvider } = useWalletStore();
+const { getChainFromChainId, getEip155ChainId } = useWalletHelper();
+const { getChainName } = useSupportedChains();
+
+function addLog(message: string, type: 'info' | 'success' | 'error' = 'info') {
+  const timestamp = new Date().toLocaleTimeString();
+  set(logs, [{ message, timestamp, type }, ...get(logs)]);
+}
+
+async function initializeWalletKit() {
+  try {
+    set(walletKit, await WalletKit.init({
+      core,
+      metadata: ROTKI_DAPP_METADATA,
+    }));
+    setupListeners();
+
+    addLog('WalletKit initialized successfully', 'success');
+  }
+  catch (error) {
+    addLog(`Failed to initialize WalletKit: ${(error as Error).message}`, 'error');
+  }
+}
+
+async function pair() {
+  const pairUriVal = get(pairUri).trim();
+  if (!pairUriVal) {
+    addLog('Please enter a valid Pair URI', 'error');
+    return;
+  }
+
+  set(isConnecting, true);
+
+  try {
+    const kit = get(walletKit);
+    if (!kit) {
```

Comment:
is there any specific reason we check twice here? also we could do the `isDefined()` and only get if it is initialized right?